src/dialog/dlgreplacecuecolor.cpp

```diff
@@ -0,0 +1,319 @@
+    }
+
+    mixxx::RgbColor::optional_t currentColor = mixxx::RgbColor::fromQString(pushButtonCurrentColor->text());
+    VERIFY_OR_DEBUG_ASSERT(currentColor) {
+        return;
+    }
+
+    int hotcueIndex = spinBoxHotcueIndex->value() - 1;
+
+    setApplyButtonEnabled(false);
+    m_dbSelectFuture = QtConcurrent::run(
+            this,
+            &DlgReplaceCueColor::selectCues,
+            *currentColor,
+            hotcueIndex,
+            conditions);
+    m_dbSelectFutureWatcher.setFuture(m_dbSelectFuture);
+}
+
+void DlgReplaceCueColor::setApplyButtonEnabled(bool enabled) {
+    QPushButton* button = buttonBox->button(QDialogButtonBox::Apply);
+    if (button) {
+        button->setEnabled(enabled);
+    }
+}
+
+QMap<int, int> DlgReplaceCueColor::selectCues(
+        mixxx::RgbColor::optional_t currentColor,
+        int hotcueIndex,
+        Conditions conditions) {
+    // The pooler limits the lifetime all thread-local connections,
+    // that should be closed immediately before exiting this function.
+    const mixxx::DbConnectionPooler dbConnectionPooler(m_pDbConnectionPool);
+    QSqlDatabase database = mixxx::DbConnectionPooled(m_pDbConnectionPool);
+
+    //Open the database connection in this thread.
+    VERIFY_OR_DEBUG_ASSERT(database.isOpen()) {
+        qWarning() << "Failed to open database for cue color replace dialog."
+                   << database.lastError();
+        return {};
+    }
+
+    //Give thread a low priority
+    QThread* thisThread = QThread::currentThread();
+    thisThread->setPriority(QThread::LowPriority);
+
+    // Build query string
+    QMap<QString, QVariant> queryValues;
+    QStringList queryStringConditions;
+    if (conditions.testFlag(ConditionFlag::CurrentColorCheck)) {
+        queryStringConditions << QString(
+                QStringLiteral("color") +
+                (conditions.testFlag(ConditionFlag::CurrentColorNotEqual) ? QStringLiteral("!=") : QStringLiteral("=")) +
+                QStringLiteral(":current_color"));
+        queryValues.insert(QStringLiteral(":current_color"), mixxx::RgbColor::toQVariant(currentColor));
+    }
+    if (conditions.testFlag(ConditionFlag::HotcueIndexCheck)) {
+        queryStringConditions << QString(
+                QStringLiteral("hotcue") +
+                (conditions.testFlag(ConditionFlag::HotcueIndexNotEqual) ? QStringLiteral("!=") : QStringLiteral("=")) +
+                QStringLiteral(":hotcue"));
+        queryValues.insert(QStringLiteral(":hotcue"), QVariant(hotcueIndex));
+    }
+
+    QString queryString = QStringLiteral("SELECT id, track_id FROM " CUE_TABLE);
+    if (!queryStringConditions.isEmpty()) {
+        queryString += QStringLiteral(" WHERE ") + queryStringConditions.join(QStringLiteral(" AND "));
+    }
+
+    // Select affected queues
+    QSqlQuery query(database);
+    query.prepare(queryString);
+    for (auto i = queryValues.constBegin(); i != queryValues.constEnd(); i++) {
```

Comment:
I believe this is equivalent to `for (const auto& val : queryValues) {}`

Reply:
No, this does not work for `QMap`.